Python/optimizer_analysis.c

```diff
@@ -436,7 +439,8 @@ remove_unneeded_uops(_PyUOpInstruction *buffer, int buffer_size)
      * instruction could have escaped. */
     int last_set_ip = -1;
     bool may_have_escaped = true;
-    for (int pc = 0; pc < buffer_size; pc++) {
+    for (int pc = 0; ; pc++) {
+        assert(pc < buffer_size && "No terminating uop");
```

Comment:
Not a huge deal, but I think I like this better the way it was. We could just stick a fatal error after the loop.